presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/TemporalCompactionSetCreator.java

```diff
@@ -96,33 +86,30 @@ public TemporalCompactionSetCreator(DataSize maxShardSize, long maxShardRows, Ty
                 consumedBytes += shard.getUncompressedSize();
                 consumedRows += shard.getRowCount();
             }
-            if (!shardsToCompact.build().isEmpty()) {
+            Set<UUID> shardUuids = shardsToCompact.build();
+            if (shardUuids.size() > 1) {
                 // create compaction set for the remaining shards of this day
-                compactionSets.add(new OrganizationSet(tableId, shardsToCompact.build(), bucketNumber));
+                compactionSets.add(new OrganizationSet(tableInfo.getTableId(), shardUuids, bucketNumber));
             }
         }
         return compactionSets.build();
     }
 
     private static class ShardSorter
```

Comment:
This can use `comparator()` from `Comparators`:
```java
return comparing(info -> info.getTemporalRange().get(),
        comparing(ShardRange::getMinTuple)
                .thenComparing(ShardRange::getMaxTuple));